client/lib/lib.rs

```diff
@@ -0,0 +1,612 @@
+    maybe_output_path: &str,
+    deploy: DeployStrParams<'_>,
+    session: SessionStrParams<'_>,
+    payment: PaymentStrParams<'_>,
+) -> Result<()> {
+    Deploy::make_deploy(
+        none_if_empty(maybe_output_path),
+        deploy.try_into()?,
+        payment.try_into()?,
+        session.try_into()?,
+    )
+}
+
+/// Transfers an amount between purses.
+///
+/// * `maybe_rpc_id` is used for RPC-ID as required by the JSON-RPC specification, and is returned
+///   by the node in the corresponding response.  It must be either able to be parsed as a `u32` or
+///   empty.  If empty, a random ID will be assigned.
+/// * `node_address` identifies the network address of the target node's HTTP server, e.g.
+///   `"http://127.0.0.1:7777"`.
+/// * When `verbose` is `true`, the request will be printed to `stdout`.
+/// * `amount` - Specifies the amount to be transferred.
+/// * `maybe_source_purse` - Source purse in the sender's account that this amount will be drawn
+///   from, if it is `""`, it will default to the account's main purse.
+/// * `maybe_target_purse` - Target purse in the sender's account that this amount will be
+///   transferred to.
+/// it is incompatible with `maybe_target_account`.
+/// * `maybe_target_account` - Target account that this amount will be transferred to.
+/// it is incompatible with `maybe_target_purse`.
+/// * `deploy` - Deploy arguments container struct holding options for this deploy.
+#[allow(clippy::too_many_arguments)]
+pub fn transfer(
+    maybe_rpc_id: &str,
+    node_address: &str,
+    verbose: bool,
+    amount: &str,
+    maybe_source_purse: &str,
+    maybe_target_purse: &str,
+    maybe_target_account: &str,
+    deploy: DeployStrParams<'_>,
+) -> Result<JsonRpc> {
+    let target = parsing::get_transfer_target(maybe_target_account, maybe_target_purse)?;
+
+    // transfer always invoke the standard payment
+    let payment = parsing::parse_standard_payment_info(amount)?;
+    let amount = U512::from_dec_str(amount)
+        .map_err(|err| Error::FailedToParseUint(UIntParseError::FromDecStr(err)))?;
+
+    let source_purse = parsing::purse::parse(maybe_source_purse).ok();
+
+    RpcCall::new(maybe_rpc_id, node_address, verbose)?.transfer(
+        amount,
+        source_purse,
+        target,
+        deploy.try_into()?,
+        payment,
+    )
+}
+
+/// Signs a deploy file.
+///
+/// * `input_path` specifies the path to the deploy file.
+/// * `secret_key` specifies the key with which to sign the deploy.
+/// * `maybe_output` specifies the output file, or if left empty, will log it to `stdout`.
+pub fn sign_deploy_file(input: &str, secret_key: &str, maybe_output: &str) -> Result<()> {
+    let input_path = parsing::input::get(input);
+    let secret_key = parsing::secret_key::parse(secret_key)?;
+    let maybe_output = parsing::output::get(maybe_output);
+    Deploy::sign_deploy_file(&input_path, secret_key, maybe_output)
+}
+
+/// Container for `Deploy` construction options.
```

Comment:
The comments on most of the member variables of `DeployStrParams`, `PaymentStrParams` and `SessionStrParams` are somewhat lacking.  These comments will be used by lib authors to understand what should be passed here, so they need to be a lot more explicit I think.

Probably referring to the help output from running `cargo r -- put-deploy --help` for each arg will give a much better indication of how these comments should look.

Reply:
Got it, thanks.